lib/holdings-updater.js

```diff
@@ -0,0 +1,295 @@
+      // If bibId is set, resolve it immediately:
+      if (subjectIdsToBibIds[subjectId]) return Promise.resolve(subjectIdsToBibIds[subjectId])
+      else {
+        // Otherwise look it up in the db:
+        return db.getStatement('resource', subjectId, 'nypl:bnum')
+          .then((result) => result.object_id.replace('urn:bnum:', ''))
+      }
+    })
+
+    return Promise.all(bibIdsToReindex)
+  }
+
+  /**
+   * This method takes a batch of inserted holding statements and 1) identifies
+   * all parent bibIds (even if bibIds not present in those statements), and
+   * 2) posts them to the configured "IndexDocumentQueue-*" stream (which
+   * triggers reindexing by the discovery-api-indexer)
+   */
+  _writeToIndexDocumentQueue (batch) {
+    let streamName = process.env.INDEX_DOCUMENT_STREAM_NAME
+    let schemaName = process.env.INDEX_DOCUMENT_SCHEMA_NAME
+    // Make sure stream-writing is configured
+    if (streamName && schemaName) {
+      this._getBibIdsForHoldingStatements(batch)
+        .then((bibIds) => {
+          let kinesisRecords = bibIds.map((bibId) => {
+            return { type: 'record', uri: bibId }
+          })
+          return this._writeToStreamsClient(streamName, kinesisRecords, schemaName)
+            .then(() => log.debug(`Wrote ${kinesisRecords.length} to ${streamName} (encoded against ${schemaName})`))
+        })
+    } else return Promise.resolve(batch)
+  }
+
+  validForSerialization (item) {
+    var valid = true
+
+    valid = valid && item.id
+
+    return valid
+  }
+
+  uriFromApi (id) {
+    return this.holdingByApi(id)
+      .catch((e) => {
+        log.info('HoldingUpdater#uriFromApi: Failed to find in api: ' + id)
+        // Fallback on cache:
+        return utils.readJson(`./cache/holdings/${id}.json`)
+          .then((holding) => {
+            log.info('HoldingUpdater#uriFromApi: Actually, found it cached: ' + holding.id)
+            return holding
+          })
+          .catch((e) => log.error('Could not find item by cache'))
+      })
+      .then((holding) => {
+        if (!holding) return Promise.reject()
+
+        log.debug('Holding: ', JSON.stringify(holding, null, 2))
+        return holding
+      })
+      .then(HoldingSierraRecord.from)
+      .then((holding) => {
+        log.info('HoldingUpdater#uriFromApi: Processing single-holding stream: [' + holding.id + ']')
+        return this._processHoldingStream(
+          H([holding])
+        )
+      })
+      .catch((e) => log.error('Error: ', e))
+  }
+
+  holdingByApi (id) {
+    log.info('HoldingUpdater#itemByApi: Fetch item by id: ', id)
+    return nyplApi.holding(id)
+      .then((holding) => {
+        if (holding) return holding
+        else return Promise.reject()
+      })
+      .catch((e) => {
+        log.error(`HoldingUpdater#itemByApi: Error fetching item from api ${id}:`)
+        return Promise.reject()
+      })
+  }
+}
+
+HoldingsUpdater.threaded = (opts) => {
```

Comment:
yeah, I honestly thought this was used for things like loading test records, etc. so its good to know that it hasn't been maintained. I'll definitely strip it out because we can much more easily push records into the `Holding` stream if/when we need to bulk load holding records.